src/scheduler.rs

```diff
@@ -51,6 +52,9 @@ pub struct Scheduler<T: Task> {
 
     // When this token is cancelled the queue has been shutdown.
     shutdown_token: CancellationToken,
+
+    // Policy for reconnection backoff when PostgreSQL connection is lost.
+    reconnection_policy: RetryPolicy,
```

Comment:
This is a nit but I think it's a bit confusing that we have `RetryPolicy`, connection policy, and task policy. Perhaps there's an opportunity for better naming, if `RetryPolicy` really is applicable to both otherwise distinct types might make better sense.

Reply:
I think I'd lean towards the second option if that reduces duplication, but I don't have a strong preference.